src/Git/GitCheckoutRevisionToTemporaryPath.php

```diff
@@ -26,10 +44,29 @@ public function checkout(CheckedOutRepository $sourceRepository, Revision $revis
 
     /**
      * {@inheritDoc}
-     * @throws RuntimeException
+     * @throws ProcessRuntimeException
      */
     public function remove(CheckedOutRepository $checkedOutRepository) : void
     {
         (new Process(['rm', '-rf', (string) $checkedOutRepository]))->mustRun();
     }
+
+    /**
+     * @throws RuntimeException
+     */
+    private function generateTemporaryPathFor(Revision $revision) : string
+    {
+        $uniquePathGenerator = $this->uniquenessFunction;
+        $checkoutDirectory   = sys_get_temp_dir() . '/api-compare-' . $uniquePathGenerator((string) $revision . '_');
+
+        if (file_exists($checkoutDirectory) || is_dir($checkoutDirectory)) {
+            throw new RuntimeException(sprintf(
+                'Tried to check out revision %s to directory %s which already exists',
```

Comment:
Put quotes around the `%s` here. Also: exception type seems to be incorrect. Can we use our own please?